src/modules/ekf2/EKF/velocity_fusion.cpp

```diff
@@ -79,12 +82,60 @@ void Ekf::updateVelocityAidSrcStatus(const uint64_t &time_us, const Vector3f &ob
 	aid_src.timestamp_sample = time_us;
 }
 
+void Ekf::fuseBodyVelocity(estimator_aid_source3d_s &aid_src)
+{
+	Vector3f body_vel_var;
+	VectorState H;
+	bool fused[3] = {false, false, false};
+	const auto state_vector = _state.vector();
+	Vector3f R(aid_src.observation_variance);
+
+	sym::ComputeEvBodyVelVarAndHx(state_vector, P, &body_vel_var, &H);
+	const Vector3f innov = _R_to_earth.transpose() * _state.vel - Vector3f(aid_src.observation);
+	innov.copyTo(aid_src.innovation);
+	(body_vel_var + R).copyTo(aid_src.innovation_variance);
+	const float innov_gate = math::max(_params.ev_vel_innov_gate, 1.f);
+	setEstimatorAidStatusTestRatio(aid_src, innov_gate);
```

Comment:
I think it would be more consistent with the other cases if you computed the test ratio prior to calling fuseBodyVelocity().

https://github.com/PX4/PX4-Autopilot/blob/612b7044515da3e107f20835ef48ba2d68719c53/src/modules/ekf2/EKF/aid_sources/external_vision/ev_vel_control.cpp#L126-L140

Reply:
Would you suggest to just move it out of the fuseBodyVelocity function?
I initially put it in there because I need to call ComputeEvBodyVelVarAndHx prior to calculate the actual test ratio.